mavisp/modules.py

```diff
@@ -105,76 +105,68 @@ def ingest(self, mutations):
-                raise MAVISpMultipleError(warning=warnings,
-                                        critical=[MAVISpCriticalError(this_error)])
-            all_methods.extend(method_dirs)
-
-        if len(all_methods) != len(set(all_methods)):
-                this_error = f"Only using one single instance of any given method is supported"
-                raise MAVISpMultipleError(warning=warnings,
-                                        critical=[MAVISpCriticalError(this_error)])
-
-        for model in models:
-
-            method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
-
-            for method_dir in method_dirs:
-                self.methods[method_dir].parse(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model, method_dir))
-                self.data = self.data.join(self.methods[method_dir].data)
-
-        keys = [ k for k in self.data.keys() if k.startswith('Stability') ]
-
-        # check if we have FoldX column
-        foldx_col = [k for k in keys if 'FoldX' in k]
-        if len(foldx_col) == 1:
-            foldx_header = foldx_col[0]
-        else:
-            foldx_header = None
-
-        if any(['Rosetta' in k for k in keys]):
-            rosetta_col = [k for k in keys if 'Rosetta' in k]
-            assert len(rosetta_col) == 1
-            rosetta_header = rosetta_col[0]
-        else:
-            rosetta_header = None
+            for model in models:
+                method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
+                if not set(method_dirs).issubset(set(self.methods.keys())):
+                    this_error = f"One or more {self.name} methods are not supported"
+                    raise MAVISpMultipleError(warning=warnings,
+                                            critical=[MAVISpCriticalError(this_error)])
+                all_methods.extend(method_dirs)
+
+                if len(all_methods) != len(set(all_methods)):
+                                this_error = f"Only using one single instance of any given method is supported"
+                                raise MAVISpMultipleError(warning=warnings,
+                                                        critical=[MAVISpCriticalError(this_error)])
+            for model in models:
+
+                method_dirs = os.listdir(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model))
+                for method_dir in method_dirs:
+                    parsed_data, this_warnings = self.methods[method_dir].parse(os.path.join(self.data_dir, self.module_dir, f'{structure_ID}_{residue_range}', method, model, method_dir))
+                    warnings += this_warnings
+                    self.data = self.data.join(parsed_data,rsuffix = f" using {method} method")
```

Comment:
you're still using `rsuffix` here so I would assume you have the same problem

also can we change the column name to something like `Stability (FoldX5, {method}, kcal/mol)`?